Tasks/js-targets.js

```diff
@@ -4,8 +4,9 @@ const eslint = require('gulp-eslint');
 
 function compile(source, destination, pipelineModifier) {
     const copyDestination = destination || source;
+    const copySource = source.endsWith('/') ? source + '**/*.js' : source;
```

Comment:
This should rather check for `endsWith('.js')` since you don't necessarily need a slash for a directory.

Reply:
That was also my first thought 😀 But you could also pass in `folder/**/*` and would not want to have the `**/*.js` added. And then I thought to myself: "Who knows what other values might make sense here?" So I went the opposite route.

To make it more fail-proof, we could check whether `source` is a directory. If it is, we'll add `**/*.js` to it, otherwise we'll leave it as it is. That check would return false for both `dir/file.js` and `dir/**/*`. What do you think?

Reply:
Or having a second function, `compileFile` would make things clearer (changing the name of this one is more of an issue because it's a published interface and it's used all over the place already).

Reply:
That's a great idea. I'll do that.

Reply:
Now, that I've written the new `compileFile` function, I'm not that convinced anymore that that's the best solution.

I think having a single `compile` method that accepts different input, as does e.g. jQuery, would be the cleaner approach, even if the inner workings of the function become more complex due to that. With the check for `source` being a directory or not, we should have most use cases covered.

I even considered allowing a stream as the `source` parameter, so you can pass in whatever you like using `gulp.src(...)` at the call site. But then I thought that would be too much.

Reply:
Let see it!

Reply:
Next week 😀